F taint propagation experiment

labrenz opened the pull request:
Allow to run vara's taint analysis on designed set of test files for propagation evaluation.

The compiled cpp files and their filecheck files can be found under [this repo](https://github.com/se-passau/vara-perf-tests/tree/f-taintTests/taint-tests).

The result of each filecheck is stored into a result file for each test file.
If the output of vara matches the expected filecheck an empty file with the `_success` file ending is created, otherwise an `_failed.txt` file will be created. The content of the failed run's output is stored in the result file.

labrenz committed:
Add local vscode settigs to git ignore.

labrenz committed:
Add minimalistic cpp examples for MTFA.

labrenz committed:
Add minimalistic cpp examples for MTFA.

labrenz committed:
Add stucture of taint_propagation experiment.

labrenz committed:
Add stucture of taint_propagation experiment.

labrenz committed:
Move taint cpp files to own project.

labrenz committed:
Move taint cpp files to own project.

labrenz committed:
Add missing imports in experiment.

labrenz committed:
Add missing imports in experiment.

labrenz committed:
Adjust clang and vara flags.

labrenz committed:
Adjust clang and vara flags.

labrenz committed:
Add binary names of new taint tests

labrenz committed:
Add binary names of new taint tests

labrenz committed:
Change binary naming to fix path for bc generation

labrenz committed:
Change binary naming to fix path for bc generation

labrenz committed:
Start on disassembly call to generate lls from project

labrenz committed:
Start on disassembly call to generate lls from project

labrenz committed:
Generate ll files for tain-test files

labrenz committed:
Generate ll files for tain-test files

labrenz committed:
Adjust vara path to MTFA generation

labrenz committed:
Adjust vara path to MTFA generation

labrenz committed:
Call printFullMTFA instead of view

labrenz committed:
Call printFullMTFA instead of view

labrenz committed:
Remove unnecessary imports

labrenz committed:
Remove unnecessary imports

labrenz committed:
Add clean step and format of results

labrenz committed:
Add clean step and format of results

labrenz committed:
Clean up directory management

labrenz committed:
Clean up directory management

labrenz committed:
Add caching for taint experiment

labrenz committed:
Add caching for taint experiment

labrenz committed:
Use bc as input for vara analysis

labrenz committed:
Use bc as input for vara analysis

labrenz committed:
Remove redundant disassemble action

labrenz committed:
Remove redundant disassemble action

labrenz committed:
Add structure of taint report

labrenz committed:
Add structure of taint report

labrenz committed:
Add structure of taint report

julianbreiteneicher committed:
Include stdout in PEErrorHandler (#113)

julianbreiteneicher committed:
Fix commit interactions plot (#115)

julianbreiteneicher committed:
Always init commits in GitBlameAnnotationReport (#114)

vulder committed:
Fixes crash in status if no config was set. (#116)

vulder committed:
Fixes sorting error in commit interaction plot (#117)

resolves se-passau/VaRA#464

vulder committed:
Fixes compilation error in xz case study (#118)

resolves se-passau/VaRA#461

vulder committed:
Remove unnecessary local copies of project folders (#119)

vulder committed:
Only analyze missing revision by default (#120)

labrenz committed:
Create filename for taintPropReport

labrenz committed:
Create filename for taintPropReport

labrenz committed:
Pipe MTFA output into TPR

labrenz committed:
Pipe MTFA output into TPR

labrenz committed:
Add more fitting module description.

labrenz committed:
Add more fitting module description.

labrenz committed:
Add structure of filechecking analysis results.

labrenz committed:
Add structure of filechecking analysis results.

labrenz committed:
Adjust file names to new perf tests.

labrenz committed:
Adjust file names to new perf tests.

labrenz committed:
Add FileCheckStep structure.

labrenz committed:
Add FileCheckStep structure.

labrenz committed:
Change back SRC_FILE of tain_tests.

labrenz committed:
Change back SRC_FILE of tain_tests.

labrenz committed:
Automatically run filecheck files on taint tests.

labrenz committed:
Automatically run filecheck files on taint tests.

labrenz committed:
Merge almost duplicated steps into one.

labrenz committed:
Merge almost duplicated steps into one.

labrenz committed:
Fixed rename that went wrong.

labrenz committed:
Fixed rename that went wrong.

labrenz committed:
Formating taint experiment.

labrenz committed:
Formating taint experiment.

labrenz committed:
Catch errors of filecheck to run all examples.

labrenz committed:
Catch errors of filecheck to run all examples.

julianbreiteneicher committed:
Bump Benchbuild version to 4.0.1 (#121)

related se-passau/VaRA#384
related se-passau/VaRA-Tool-Suite#111

vulder committed:
Disable shallow git clone for currently used case study projects (#111)

labrenz committed:
Enable custom file types for successful reports (#112)

Allows different reports to specify their own file type extension, while vara-cs still recognizes them as successful runs and paints their statuses accordingly.

To migrate old files to the new format use(for rename from linux utils):
cd vara/results/
rename '.yaml' '_success.yaml' **/*
rename '.failed' '_failed.txt' **/*
rename '.cerror' '_cerror.txt' **/*

or the sed style for debian rename
rename 's/.yaml/_sucess.yaml/' **/*

labrenz committed:
Enable custom file types for successful reports (#112)

Allows different reports to specify their own file type extension, while vara-cs still recognizes them as successful runs and paints their statuses accordingly.

To migrate old files to the new format use(for rename from linux utils):
cd vara/results/
rename '.yaml' '_success.yaml' **/*
rename '.failed' '_failed.txt' **/*
rename '.cerror' '_cerror.txt' **/*

or the sed style for debian rename
rename 's/.yaml/_sucess.yaml/' **/*

labrenz committed:
Enable custom file types for successful reports (#112)

Allows different reports to specify their own file type extension, while vara-cs still recognizes them as successful runs and paints their statuses accordingly.

To migrate old files to the new format use(for rename from linux utils):
cd vara/results/
rename '.yaml' '_success.yaml' **/*
rename '.failed' '_failed.txt' **/*
rename '.cerror' '_cerror.txt' **/*

or the sed style for debian rename
rename 's/.yaml/_sucess.yaml/' **/*

labrenz committed:
Use custom file extension for failed runs.

labrenz committed:
Use custom file extension for failed runs.

julianbreiteneicher committed:
Increates projetcs' sample limit to 200 (#122)

julianbreiteneicher committed:
Fix missing/empty cerror experiment report files (#123)

vulder committed:
Fixes wrong function name, removing typo (#124)

julianbreiteneicher committed:
Revert to matplotlib 3.1.0 until 3.1.2 is released (#126)

The current matplotlib version 3.1.1 breaks seaborn heatmaps.
Matplotlib 3.1.2 will fix this issue, but it is not released yet.
Once it is released, the setup.py entry should be changed to
"matplotlib>=3.1.2".

labrenz committed:
Set error file extension to .txt (#125)

julianbreiteneicher committed:
Fix case-study overview plot (#127)

labrenz committed:
Formatings to cmd chain, not fixed yet.

labrenz committed:
Formatings to cmd chain, not fixed yet.

labrenz committed:
Formatings to cmd chain, not fixed yet.

labrenz committed:
Fix generation of success files on error.

labrenz committed:
Fix generation of success files on error.

Commit:
Fixes cmake phasar variable name (#130)

vulder committed:
Fixes terminal line reset for empty output (#132)

vulder committed:
Updates setup doc with VaRA deps (#131)

vulder committed:
Adds all mode to the dev status tool. (#133)

resolves se-passau/VaRA#483

vulder commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- varats/data/reports/taint_report.py  1
- varats/projects/test_projects/taint_tests.py  3
- varats/experiments/taint_propagation.py  4
         

Clones added
============
- varats/projects/c_projects/lrzip.py  1
- varats/experiments/git_blame_annotation_report.py  2
- varats/projects/c_projects/libvpx.py  1
- varats/experiments/taint_propagation.py  2
         

Clones removed
==============
+ varats/projects/c_projects/xz.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4280681&bid=14603033)

labrenz commented:
Closing again, while i correctly resolve all merging issues with the vara-dev branch.

labrenz committed:
Fix typing and commit report failed ending.

labrenz committed:
Merge branch 'f-TaintPropagationExperiment' of github.com:se-passau/VaRA-Tool-Suite into f-TaintPropagationExperiment

labrenz commented:
Conflicts should all be resolved now.

vulder commented on varats/data/reports/taint_report.py:
```diff
@@ -0,0 +1,49 @@
+    in a readable manner.
+    """
+
+    SHORTHAND = "TAINT"
```
maybe we should use something more specific here, TAINTPROP or short TP? or maybe `TPR` ?

vulder replied:
you used `TPR` later in the `__repr__` impl maybe this is the best solution

vulder commented on varats/data/reports/taint_report.py:
```diff
@@ -0,0 +1,49 @@
+
+    def __init__(self, path: Path) -> None:
+        super(TaintPropagationReport, self).__init__()
+        with open(path, "r") as stream:
```
why do you open the file as a `stream` but do not use it?

labrenz replied:
Resolved in commit 87f44bdb0f23d4317a52197ddc4f6e83ee21949b.

vulder commented on varats/data/reports/taint_report.py:
```diff
@@ -0,0 +1,49 @@
+    def __init__(self, path: Path) -> None:
+        super(TaintPropagationReport, self).__init__()
+        with open(path, "r") as stream:
+                self._path = path
```
```suggestion
                self.__path = path
```
private vars should have two underscores

vulder commented on varats/data/reports/taint_report.py:
```diff
@@ -0,0 +1,49 @@
+        """
+        Path to TaintPropagationReport file.
+        """
+        return self._path
```
```suggestion
        return self.__path
```

labrenz replied:
Resolved in commit 87f44bdb0f23d4317a52197ddc4f6e83ee21949b.

vulder commented on varats/data/reports/taint_report.py:
```diff
@@ -0,0 +1,49 @@
+                                        extension_type, file_ext)
+
+    def __repr__(self) -> str:
+        return "TPR: " + os.path.basename(self.path)
```
```suggestion
        return self.SHORTHAND + ": " + os.path.basename(self.path)
```

vulder commented on varats/experiments/taint_propagation.py:
```diff
@@ -0,0 +1,271 @@
+        timeout_duration = '8h'
+
+        for binary_name in project.BIN_NAMES:
+
```
```suggestion
```

labrenz replied:
Resolved in commit 87f44bdb0f23d4317a52197ddc4f6e83ee21949b.

vulder commented on varats/experiments/taint_propagation.py:
```diff
@@ -0,0 +1,271 @@
+        if not all_cache_files_present:
+            analysis_actions.append(actions.Compile(project))
+            analysis_actions.append(Extract(project))
+            analysis_actions.append(FileCheckExpected(project))
```
Is it really needed to cache the filecheck outputs? Normally only generating the `.bc` files is expensive, reanalyzing a already generated `.bc` file should be cheap in comparison.

labrenz replied:
Resolved in commit 87f44bdb0f23d4317a52197ddc4f6e83ee21949b.

Removed the txt files from the BC_FILES directory and make use of benchbuilds tmp directory.

labrenz committed:
Remove caching of .txt files.

labrenz committed:
Merge branch 'vara-dev' into f-TaintPropagationExperiment

labrenz committed:
Add missing ',' in settings.

labrenz reviewed (commented):
Reviewed further changes, applied requested feedback and merged the new vara-dev.

labrenz committed:
Adjust module description.

labrenz committed:
Remove filecheck from the first vara taint experiment.

The filecheck will be added again in an additional experiment, that
extends this experiment.

labrenz committed:
Readd the exception handling.

labrenz committed:
Once again remove exceptiom block and an import.

labrenz committed:
Merge remote-tracking branch 'origin/vara-dev' into f-TaintPropagationExperiment

labrenz committed:
Add parsing and filecheck exp for vara,

labrenz committed:
Merge pull request #143 from se-passau/f-FileCheckVaRATaints

Add parsing and filecheck exp for vara,

vulder commented on varats/data/reports/taint_report.py:
```diff
@@ -0,0 +1,48 @@
+                                        extension_type, file_ext)
+
+    def __repr__(self) -> str:
+        return self.SHORTHAND + ":" + os.path.basename(self.path)
```
```suggestion
        return self.SHORTHAND + ": " + os.path.basename(self.path)
```
I normally find a space after a colon easier to comprehend and copy. If you want it without or if the formatting is necessary, it's fine with me.

vulder commented on varats/experiments/vara_full_mtfa.py:
```diff
@@ -0,0 +1,179 @@
+    """
+
+    NAME = "VaRATaintPropagation"
+
```
```suggestion
```

vulder reviewed (approved):
Please fix the two nits otherwise it's fine, I'll merge it afterwards.

labrenz committed:
Update varats/data/reports/taint_report.py

Co-Authored-By: Florian Sattler <vuld3r@gmail.com>

labrenz committed:
Remove empty line.

Co-Authored-By: Florian Sattler <vuld3r@gmail.com>
